arboretum/kubernetes/fetchers/fetch_cluster_list.py

```diff
@@ -0,0 +1,37 @@
+# -*- mode:python; coding:utf-8 -*-
+# Copyright (c) 2020 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Fetch cluster list by using other providers' fetch_cluster_list function."""
+import json
+
+from compliance.evidence import store_raw_evidence
+from compliance.fetch import ComplianceFetcher
+
+
+class ClusterListFetcher(ComplianceFetcher):
+    """Fetch BOM (Bill of Materials) as cluster list."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Initialize the fetcher object with configuration settings."""
+        cls.logger = cls.locker.logger.getChild(
+            'kubernetes.cluster_list_fetcher'
+        )
+        return cls
+
+    @store_raw_evidence('kubernetes/cluster_list.json')
+    def fetch_cluster_list(self):
+        """Fetch BOM (Bill of Materials) as cluster list."""
+        bom = self.config.get('org.kubernetes.cluster_list.bom')
+        return json.dumps(bom)
```

Comment:
I don't understand, this fetcher doesn't seem to do anything?

Reply:
As you mentioned, this fetcher just copies the list of clusters into an evidence file.  

According to the discussion with @drsm79, we plan to simplify the cluster *resource* fetcher by unifying the format and the source of cluster list.  This simple fetcher makes an evidence locker as an only one source of cluster list, and the cluster resource fetcher can read the cluster list in a unified way (regardless of cluster types).

@drsm79 Please supplement my comment if I'm not describing our design principle of the cluster list/resource fetchers correctly.

Reply:
Yeah, a BOM is "just" configuration. We _could_ do this via plant, but I think the running config is a better source. Willing to be persuaded otherwise, though.

Reply:
@alfinkel just reminded me of https://github.com/ComplianceAsCode/auditree-arboretum/blob/dcb2d2c5ae419e465c4b229f473d7175beeda845/arboretum/auditree/fetchers/fetch_compliance_config.py#L23 so I think we document the configuration your downstream code expects, remove this fetcher and everything is ticketty boo. Right?